packages/ui/src/hooks/usePaginate.ts

```diff
@@ -0,0 +1,34 @@
+import { useState } from 'react';
+
+const DEFAULT_ITEMS_PER_PAGE = 5;
+
+export const usePaginate = (
+  pageSize = DEFAULT_ITEMS_PER_PAGE,
+  listLength?: number,
+) => {
+  const [activePage, setActivePage] = useState(0);
+  const paginate = <T>(arr: T[]) => {
+    const startIndex = activePage * pageSize;
+    const endIndex =
+      arr?.length && startIndex + pageSize > arr?.length
+        ? arr?.length
+        : startIndex + pageSize;
+    return arr?.slice(startIndex, endIndex);
+  };
+
+  const pageCount =
+    listLength !== undefined ? Math.ceil(listLength / pageSize) : undefined;
+
+  return {
+    activePage,
+    setActivePage,
+    paginate,
+    canNextPage: pageCount === undefined || activePage < pageCount - 1,
```

Comment:
Naming for `canNextPage` and `canPreviousPage` is weird and you always use it with negation (e.g. !canNextPage) to find out if a button is `disabled`. Just name it `isNextButtonDisabled` and `isPreviousButtonDisabled` and it will be much cleaner.

Also, use `useMemo`.